frontend/src/App.jsx

```diff
@@ -69,6 +72,35 @@ function App() {
   const closeModal = () => {
     setShowModal(false);
   };
+  
+  useEffect(() => {
+    const checkMobile = () => {
+      const userAgent = navigator.userAgent || navigator.vendor || window.opera;
+      
+      const mobileRegex = /android|webos|iphone|ipad|ipod|blackberry|iemobile|opera mini/i;
+      
+      const isMobileDevice = mobileRegex.test(userAgent);
+      setIsMobile(isMobileDevice);
+
+      const isMobileWidth = window.innerWidth <= 768;
+      setIsMobile(isMobileDevice || isMobileWidth);
+    };
+
+    checkMobile();
+
+    window.addEventListener('resize', checkMobile);
+
+    return () => window.removeEventListener('resize', checkMobile);
+  }, []);
+
+  const handleCloseModal = () => {
+    setModal(false);
+  };
+
+  const openTelegramBot = () => {
+    const telegramBotLink = 'https://t.me/spotnet_bot';
```

Comment:
extract it into `frontend/src/utils/constants.js`, rename to be in constant style (uppercase) as well